src/useradd.c

```diff
@@ -2203,8 +2203,13 @@ static void create_mail (void)
 		mode_t mode;
 
 		spool = getdef_str ("MAIL_DIR");
+#ifdef MAIL_SPOOL_DIR
+		if ((NULL == spool) && (getdef_str ("MAIL_FILE") == NULL)) {
+			spool = MAIL_SPOOL_DIR;
+		}
+#endif /* MAIL_SPOOL_DIR */
 		if (NULL == spool) {
-			spool = "/var/mail";
+			return;
```

Comment:
MAIL_FILE, according to login.defs manpage,
    Defines the location of the users mail spool files relatively to their home directory.
So for userdel , returning here makes sense, as $HOME will be deleted anyway.
But if MAIL_FILE is defined, we should probably create it here,
not exit early.  Right?

Reply:
I'm not the author of the patch but found the same issue. So far useradd never did anything with MAIL_FILE. So the patch can't break that :-) The patch does fix an issue though. On systems that do not have /var/mail, useradd prints an error message at the moment, even when MAIL_SPOOL_DIR was intentionally either not set during build or detected to be a different location and /var/mail not a symlink pointing there.

Reply:
thanks.